### PR TITLE
feature(project): #1227 added index file for models and pipes

### LINKED ISSUE #1227: Services, pipes, models to use index file

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1227 

### Notes

A summary of what was achieved in this PR

- [ ] Updated index.model file for navigation and user model
- [ ] Added index file for pipes and updated its referenced

### CODE REVIEW

**Comment by webkhushboo** [comment]
@eddiejaoude 
General question: Why do we need index file for services? If I am adding index file then importing the services using index.service file . 
Its causing circular dependency issues
repostory.service.ts -- > index.service.ts --> repository.service.ts file 

Please let me know your reviews on it

**Comment by eddiejaoude** [comment]
The benefit of the index file, is that only a single import is required, for example

```typescript
import { DigitsPipe, UserPipe, EnvironmentTypePipe } from './index.pipe';
```

**Comment by eddiejaoude** [comment]
Should not get a circular reference, it is technically the same.

It goes from...
```
service -> modelA
service -> modelB
service -> modelC
```

to

```
service -> model index -> modelA / modelB / modelC
```

**Comment by eddiejaoude** [comment]
> Its causing circular dependency issues

Yes, sorry, just realised, anything in the same directory does not use `index`. For example, `serviceA` would import `serviceB` but a component would use the `index`

**Comment by webkhushboo** [comment]
@eddiejaoude 
Do we really need index file for services ?

**Comment by eddiejaoude** [comment]
Yep, the same again. Any file in service directory does not use the `index.services`, but everywhere else does. Then you won't get the circular reference

**Comment by webkhushboo** [comment]
You mean in repository.service file we will be importing authentication.service.ts directly ?

**Comment by eddiejaoude** [comment]
Yep 👍 . Any directory that has an `index`, those file don't use the `index` themselves.

**Comment by webkhushboo** [comment]
great 👍

File: web/src/app/shared/models/profile.model.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ActivityModel } from './activity.model';
+import { ActivityModel } from './index.model';
```
Comment (eddiejaoude) [inline web/src/app/shared/models/profile.model.ts]: The models themselves import the files directly. This change is not required

File: web/src/app/shared/models/event.model.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
-import { UserModel } from './user.model';
-import { RepositoryModel } from './repository.model';
+import { RepositoryModel, UserModel } from './index.model';
```
Comment (eddiejaoude) [inline web/src/app/shared/models/event.model.ts]: These 2 original imports are correct

File: web/src/app/shared/models/activity.model.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { RepositoryModel } from './repository.model';
+import { RepositoryModel } from './index.model';
```
Comment (eddiejaoude) [inline web/src/app/shared/models/activity.model.ts]: Same here, original import is fine as its loading files in the same type/directory (in this case `model`s)

File: web/src/app/core/services/repository.service.ts [hunk 4]
```diff
@@ -10,8 +10,7 @@ import { pluck } from 'rxjs/operators';
 
 // Dashboard hub model and services
 import { AuthenticationService } from './authentication.service';
-import { RepositoriesModel } from '../../shared/models/index.model';
-import { RepositoryModel, ProfileModel } from '../../shared/models/index.model';
+import { RepositoryModel, RepositoriesModel, ProfileModel } from '../../shared/models/index.model';
```
Comment (eddiejaoude) [inline web/src/app/core/services/repository.service.ts]: Looks good 👍

**Comment by eddiejaoude** [comment]
Looks good 👍 , a few comments where I think `index` should not be used as its in the same directory (type), for example model importing another model should be direct and not via the `index`

File: web/src/app/core/services/index.service.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export * from './authentication.service';
+export * from './project.service';
+export * from './repository.service';
+export * from './user.service';
+export * from './spinner.service';
```
Comment (eddiejaoude) [inline web/src/app/core/services/index.service.ts]: `spinner` should be before `user`

Comment (webkhushboo) [reply web/src/app/core/services/index.service.ts]: done

File: web/src/app/core/services/repository.service.ts (hunk 4 above)
Comment (webkhushboo) [reply web/src/app/core/services/repository.service.ts]: Thanks

File: web/src/app/shared/models/activity.model.ts (hunk 3 above)
Comment (webkhushboo) [reply web/src/app/shared/models/activity.model.ts]: done

File: web/src/app/shared/models/event.model.ts (hunk 2 above)
Comment (webkhushboo) [reply web/src/app/shared/models/event.model.ts]: done

File: web/src/app/shared/models/profile.model.ts (hunk 1 above)
Comment (webkhushboo) [reply web/src/app/shared/models/profile.model.ts]: done

**Pull request merged**